source/common/common/thread.cc

```diff
@@ -21,7 +25,15 @@ Thread::Thread(std::function<void()> thread_routine) : thread_routine_(thread_ro
   UNREFERENCED_PARAMETER(rc);
 }
 
-int32_t Thread::currentThreadId() { return syscall(SYS_gettid); }
+int32_t Thread::currentThreadId() {
+#ifdef __linux__
+  return syscall(SYS_gettid);
+#elif defined(__APPLE__)
+  int ret = mach_thread_self();
+  mach_port_deallocate(mach_task_self(), ret);
```

Comment:
Sorry I didn't catch this the first time. I found some info while searching about mach_thread_self/mach_port_deallocate:

https://codereview.chromium.org/276043002/

Did you try `pthread_mach_thread_np(pthread_self())`?